lib/diff.ml

```diff
@@ -110,6 +135,16 @@ and module_declaration ~typing_env ~name ~reference ~current =
       modtype_item ~loc:reference.md_loc ~typing_env ~name
         ~reference:ref_modtype ~current:curr_modtype
 
+and module_type_declaration ~typing_env ~name ~reference ~current =
```

Comment:
This function has some duplicated code with `module_declaration`, could you factor things to avoid this duplication?